src/node/rpc/common_handler_registry.h

```diff
@@ -68,6 +61,67 @@ namespace ccf
           "Failed to get local commit info from Consensus");
       };
 
+      auto get_tx_status = [this](Store::Tx& tx, nlohmann::json&& params) {
+        const auto in = params.get<GetTxStatus::In>();
+
+        if (consensus != nullptr)
+        {
+          const auto actual_term = consensus->get_view(in.commit);
+
+          GetTxStatus::Out out;
+
+          if (actual_term == 0)
+          {
+            // This node has not seen this index yet - assume transaction is
+            // still pending (in-flight)
+
+            // NB: There are subtleties here around elections. It is tempting to
+            // say that if the local view has advanced past the target view,
+            // then the target version is Lost. This node may be mid-election,
+            // where the eventual winner has this version and will successfully
+            // replicate it. For safety, we consider all such transactions to be
+            // pending. This means they may be pending indefinitely
+            // (requesting 1.8, the service is stable with no new transactions
+            // at 2.5)!
+            out.status = GetTxStatus::TxStatus::Pending;
+          }
+          else if (actual_term < in.term)
+          {
+            // The requested transaction was committed in an _earlier_ view?
+            // Should be impossible for queries about real assigned versions,
+            // but essentially the same as the case below ("this version does
```

Comment:
Minor: "case below" would no longer apply if the order of this if/else gets changed. Better to refer to actual case explicitly?

Reply:
I've rewritten this logic, now in `tx_status.h`, hopefully with clearer decision path.